documentation/source/newsfragments/1381.removal.rst

```diff
@@ -0,0 +1 @@
+String hierarchy objects now expect and resolve to :class:`bytes` objects, rather than taking :class:`str` objects and assuming an encoding scheme. Users are now expected to choose an encoding when using such objects. As a convience, and to prevent breakage of currently functioning tests, when assinging :class:`str` values to a string heirarchy object, the ASCII encoding scheme will be used as a safe default. This convenience will be removed after 1.4.
```

Comment:
```suggestion
The value of :class:`cocotb.handle.StringObject`\ s is now of type :class:`bytes`, instead of  :class:`str` objects with an implied utf8 encoding scheme. Users are now expected to choose an encoding when using such objects. As a convience, and to prevent breakage of currently functioning tests, when assigning :class:`str` values, the ASCII encoding scheme will be used as a safe default.
```